### PR TITLE
gather-extra: Add list of all ports

### PR DESCRIPTION
**Pull request by jsafrane**
List all ports on all nodes to networking/ss-*. This will be used to debug https://bugzilla.redhat.com/show_bug.cgi?id=1990662 (i.e. to see who's occupying CSI driver ports), but it may be useful also for other purposes.

Fix existing networking/iptables-save when at it (`-timeout` is not a valid oc argument)

### CODE REVIEW

**Comment by jsafrane** [comment]
Sample output: https://gcsweb-ci.apps.ci.l2s4.p1.openshiftapps.com/gcs/origin-ci-test/pr-logs/pull/openshift_release/24908/rehearse-24908-pull-ci-openshift-csi-driver-shared-resource-release-4.11-e2e-aws-csi-driver/1478295812562751488/artifacts/e2e-aws-csi-driver/gather-extra/artifacts/network/

File: ci-operator/step-registry/gather/extra/gather-extra-commands.sh [hunk 1]
```diff
@@ -140,7 +140,12 @@ done
 # Snapshot iptables-save on each node for debugging possible kube-proxy issues
 oc --insecure-skip-tls-verify get --request-timeout=20s -n openshift-sdn -l app=sdn pods --template '{{ range .items }}{{ .metadata.name }}{{ "\n" }}{{ end }}' > /tmp/sdn-pods
 while IFS= read -r i; do
-  queue ${ARTIFACT_DIR}/network/iptables-save-$i oc --insecure-skip-tls-verify rsh --timeout=20 -n openshift-sdn -c sdn $i iptables-save -c
+  queue ${ARTIFACT_DIR}/network/iptables-save-$i oc --insecure-skip-tls-verify rsh --request-timeout=20s -n openshift-sdn -c sdn $i iptables-save -c
+done < /tmp/sdn-pods
+
+# Snapshot all used ports on each node.
+while IFS= read -r i; do
+  queue ${ARTIFACT_DIR}/network/ss-$i oc --insecure-skip-tls-verify rsh --request-timeout=20s -n openshift-sdn -c sdn $i ss -apn
```
Comment (tssurya) [inline ci-operator/step-registry/gather/extra/gather-extra-commands.sh:148]: Is this something we should be doing for `openshift-ovn-kubernetes` as well? I know we haven't exactly been gathering the iptables for ovn-k though that would be great... but its probably not exactly related to this PR unless the bug https://bugzilla.redhat.com/show_bug.cgi?id=1990662 happens irrespective of the networking plugin?

**Review by tssurya — COMMENTED** [review]
logic lgtm (waiting for CI to pass)
this fixes the `--timeout` flag which is wrong to `--request-timeout` &
adding the `ss -anp` command to list all ports per node.

File: ci-operator/step-registry/gather/extra/gather-extra-commands.sh (hunk 1 above)
Comment (jsafrane) [reply ci-operator/step-registry/gather/extra/gather-extra-commands.sh]: I think my BZ is not related to any particular network plugin - how would a network plugin affect which ports on nodes are used?
I reworked the code a bit to collect network info from both OVN and SDN pods, one of them should succeed.

File: ci-operator/step-registry/gather/extra/gather-extra-commands.sh [hunk 2]
```diff
@@ -137,11 +137,31 @@ for path in "${paths[@]}" ; do
...
+  done < ${podlist}
+  # Snapshot all used ports on each node.
+  while IFS= read -r i; do
+    queue ${ARTIFACT_DIR}/network/ss-$i oc --insecure-skip-tls-verify rsh --request-timeout=20s -n ${namespace} -c ${container} $i ss -apn
+  done < ${podlist}
+}
+
+# Gather network details both from SDN and OVN. One of them should succeed.
+gather_network openshift-sdn app=sdn sdn
+gather_network openshift-ovn-kubernetes app=ovnkube-node ovnkube-node
```
Comment (tssurya) [inline ci-operator/step-registry/gather/extra/gather-extra-commands.sh:164]: ++

**Review by tssurya — COMMENTED** [review]
/lgtm

@jsafrane : PR lgtm, please double check why the specific jobs are failing in the rehearsal part of CI, I haven't dug deep.

I thought we had a "gather-network" somewhere where this would be a better fit, but it turns out that's a part of must-gather and so I am fine if this is put in the gather-extra for now since the iptables-save was already here.

**Comment by tssurya** [comment]
output looks good: https://gcsweb-ci.apps.ci.l2s4.p1.openshiftapps.com/gcs/origin-ci-test/pr-logs/pull/openshift_release/24908/rehearse-24908-pull-ci-openshift-openshift-tests-private-release-4.11-e2e-console/1479096712629850112/artifacts/e2e-console/gather-extra/artifacts/network/

**Comment by jsafrane** [comment]
rebased

**Comment by jsafrane** [comment]
Sample of an OVN job: https://gcsweb-ci.apps.ci.l2s4.p1.openshiftapps.com/gcs/origin-ci-test/pr-logs/pull/openshift_release/24908/rehearse-24908-pull-ci-openshift-cluster-etcd-operator-release-4.9-e2e-gcp-disruptive-ovn/1481222488745578496/artifacts/e2e-gcp-disruptive-ovn/gather-extra/artifacts/network/
SDN job: https://gcsweb-ci.apps.ci.l2s4.p1.openshiftapps.com/gcs/origin-ci-test/pr-logs/pull/openshift_release/24908/rehearse-24908-pull-ci-openshift-cluster-storage-operator-release-4.11-e2e-azure-file-csi/1481222488854630400/artifacts/e2e-azure-file-csi/gather-extra/artifacts/network/

**Comment by tssurya** [comment]
looking good:  https://gcsweb-ci.apps.ci.l2s4.p1.openshiftapps.com/gcs/origin-ci-test/pr-logs/pull/openshift_rele[…]or-disconnected/gather-extra/artifacts/network/

**Review by vrutkovs — APPROVED** [review]
/approve

**Pull request merged**